### PR TITLE
IOS/DI: Fake the error 001 read when running DirectoryBlobs or Riivolution-patched games.

### PR DESCRIPTION
**Pull request by AdmiralCurtiss**
@JosJuice Is this what you had in mind?

### CODE REVIEW

File: Source/Core/Core/BootManager.cpp [hunk 1]
```diff
@@ -144,6 +145,17 @@ bool BootCore(std::unique_ptr<BootParameters> boot, const WindowSystemInfo& wsi)
...
+  // DirectoryBlobs (including Riivolution-patched discs) may end up larger than a real physical Wii
+  // disc, which triggers Error #001. In those cases we manually make the check succeed to avoid
+  // problems.
+  const bool should_fake_error_001 =
+      StartUp.bWii &&
+      (!boot->riivolution_patches.empty() ||
+       (std::holds_alternative<BootParameters::Disc>(boot->parameters) &&
+        std::get<BootParameters::Disc>(boot->parameters).volume->GetBlobReader().GetBlobType() ==
+            DiscIO::BlobType::DIRECTORY));
+  Config::SetCurrent(Config::SESSION_SHOULD_FAKE_ERROR_001, should_fake_error_001);
```
Comment (JosJuice) [inline Source/Core/Core/BootManager.cpp:157]: Ideally this should be implemented in such a way that if the user switches discs, the hack turns on or off to match the new disc.

**Review by JosJuice — COMMENTED** [review]
Not sure if we should do this for all DirectoryBlobs... It's blatantly a hack, but then again, the DirectoryBlobs we create never match "real" ISOs 100% anyway.

File: Source/Core/Core/BootManager.cpp (hunk 1 above)
Comment (AdmiralCurtiss) [reply Source/Core/Core/BootManager.cpp:157]: Oh right, we do support mid-game disc swapping...

**Comment by AdmiralCurtiss** [comment]
I've enabled it for DirectoryBlobs because I've run into this before while modding -- left a backup copy of an archive in the folder and suddenly it didn't boot anymore. And since that seems like the primary use case for DirectoryBlobs I imagine I'm not the first person to run into that.

**Comment by AdmiralCurtiss** [comment]
@JosJuice So what are we doing with this? IMO this is an acceptable hack considering how DirectoryBlob works.

**Pull request merged**